### ISSUE TITLE
Suggestion: Include the Product Field in the API Response

### ISSUE THREAD

**Issue by SruiSan** [cvefeed/cvefeed.io#1]
Hi,

First of all, congratulations on the great work with CVEFeed.io. The tool is incredibly valuable, and the attention to detail in its design and functionality is outstanding.

I’d like to make a small suggestion regarding the API response for calls made to https://cvefeed.io/api/user/products/cve-feed. It would be immensely helpful if the JSON response could include the product associated with the vulnerability.

For users subscribed to multiple products, this additional field would allow us to distinguish vulnerabilities by product and implement automated workflows or actions based on this field. It would significantly enhance the usability of the API for automation purposes.

Thank you again for your hard work, and I hope this suggestion can be considered for a future update!

Best regards

**Comment by cvefeed** [comment]
Hey @SruiSan ,

Thank you very much for your kind words. 

The API you have mention is designed to be used by UI of the cvefeed.io . I'll be releasing a fully function REST API for such a operations and information soon. 

But instead of keep you waiting for a long time. I'll do my best to give this request the highest priority on my to-do list !

Thank you again for the feedback.

**Comment by cvefeed** [comment]
@SruiSan 

I finally have free time to rollout this feature you want. Now you must see the details affected_products field of the cves.

```
{
  "count": 42273,
  "next": "https://cvefeed.io/api/user/products/cve-feed?page=2",
  "previous": null,
  "results": [
    {
      "id": "CVE-2023-52922",
      "title": "CVE-2023-52922 - \"BCM Proc Show UAF Vulnerability\"",
      "description": "In the Linux kernel, the following vulnerability has been resolved:\n\ncan: bcm: Fix UAF in bcm_proc_show()\n\nBUG: KASAN: slab-use-after-free in bcm_proc_show+0x969/0xa80\nRead of size 8 at addr ffff888155846230 by task cat/7862\n\nCPU: 1 PID: 7862 Comm: cat Not tainted 6.5.0-rc1-00153-gc8746099c197 #230\nHardware name: QEMU Standard PC (i440FX + PIIX, 1996), BIOS 1.15.0-1 04/01/2014\nCall Trace:\n \u003CTASK\u003E\n dump_stack_lvl+0xd5/0x150\n print_report+0xc1/0x5e0\n kasan_report+0xba/0xf0\n bcm_proc_show+0x969/0xa80\n seq_read_iter+0x4f6/0x1260\n seq_read+0x165/0x210\n proc_reg_read+0x227/0x300\n vfs_read+0x1d5/0x8d0\n ksys_read+0x11e/0x240\n do_syscall_64+0x35/0xb0\n entry_SYSCALL_64_after_hwframe+0x63/0xcd\n\nAllocated by task 7846:\n kasan_save_stack+0x1e/0x40\n kasan_set_track+0x21/0x30\n __kasan_kmalloc+0x9e/0xa0\n bcm_sendmsg+0x264b/0x44e0\n sock_sendmsg+0xda/0x180\n ____sys_sendmsg+0x735/0x920\n ___sys_sendmsg+0x11d/0x1b0\n __sys_sendmsg+0xfa/0x1d0\n do_syscall_64+0x35/0xb0\n entry_SYSCALL_64_after_hwframe+0x63/0xcd\n\nFreed by task 7846:\n kasan_save_stack+0x1e/0x40\n kasan_set_track+0x21/0x30\n kasan_save_free_info+0x27/0x40\n ____kasan_slab_free+0x161/0x1c0\n slab_free_freelist_hook+0x119/0x220\n __kmem_cache_free+0xb4/0x2e0\n rcu_core+0x809/0x1bd0\n\nbcm_op is freed before procfs entry be removed in bcm_release(),\nthis lead to bcm_proc_show() may read the freed bcm_op.",
      "published": "2024-11-28T15:15:17.260000Z",
      "last_modified": "2024-11-28T15:15:17.260000Z",
      "cvss_score": 0,
      "severity": "NA",
      "cvss_version": null,
      "epss_scores": [],
      "url": "https://cvefeed.io/vuln/detail/CVE-2023-52922",
      "cisa_required_action": null,
      "cisa_exploit_added": null,
      "source_identifier": "416baaa9-dc9f-4396-8d5f-8c081fb06d67",
      "affected_products": [
        {
          "id": 69,
          "name": "Linux kernel",
          "vendor": "Linux",
          "get_absolute_url": "/vuln/product/69/linuxlinux_kernel/",
          "vendor_part": "o"
        }
      ]
    },
    {
      "id": "CVE-2018-9377",
      "title": "CVE-2018-9377 - Apache Http Server Information Disclosure Vulnerability",
      "description": "In BnAudioPolicyService::onTransact of IAudioPolicyService.cpp, there is a possible information disclosure due to uninitialized data. This could lead to local information disclosure with no additional execution privileges needed. User interaction is not needed for exploitation.",
      "published": "2024-11-28T01:15:04.320000Z",
      "last_modified": "2024-11-29T22:15:06.483000Z",
      "cvss_score": 6.2,
      "severity": "MEDIUM",
      "cvss_version": "cvss31",
      "epss_scores": [],
      "url": "https://cvefeed.io/vuln/detail/CVE-2018-9377",
      "cisa_required_action": null,
      "cisa_exploit_added": null,
      "source_identifier": "security@android.com",
      "affected_products": [
        {
          "id": 2,
          "name": "Android",
          "vendor": "Google",
          "get_absolute_url": "/vuln/product/2/googleandroid/",
          "vendor_part": "o"
        }
      ]
    },
    {
      "id": "CVE-2018-9374",
      "title": "CVE-2018-9374 - Apache PackageManager Local Privilege Escalation Vulnerability",
      "description": "In installPackageLI of PackageManagerService.java, there is a possible permissions bypass. This could lead to local escalation of privilege with User execution privileges needed. User interaction is not needed for exploitation.",
      "published": "2024-11-28T00:15:03.827000Z",
      "last_modified": "2024-11-29T22:15:06.327000Z",
      "cvss_score": 7.8,
      "severity": "HIGH",
      "cvss_version": "cvss31",
      "epss_scores": [],
      "url": "https://cvefeed.io/vuln/detail/CVE-2018-9374",
      "cisa_required_action": null,
      "cisa_exploit_added": null,
      "source_identifier": "security@android.com",
      "affected_products": [
        {
          "id": 2,
          "name": "Android",
          "vendor": "Google",
          "get_absolute_url": "/vuln/product/2/googleandroid/",
          "vendor_part": "o"
        }
      ]
    },
    {
      "id": "CVE-2018-9354",
      "title": "CVE-2018-9354 - VendorOne VideoFrame Denial-of-Service",
      "description": "In VideoFrameScheduler.cpp of VideoFrameScheduler::PLL::fit, there is a possible remote denial of service due to divide by 0. This could lead to remote denial of service with no additional execution privileges needed. User interaction is needed for exploitation.",
      "published": "2024-11-27T23:15:04.650000Z",
      "last_modified": "2024-11-29T22:15:06.180000Z",
      "cvss_score": 6.5,
      "severity": "MEDIUM",
      "cvss_version": "cvss31",
      "epss_scores": [],
      "url": "https://cvefeed.io/vuln/detail/CVE-2018-9354",
      "cisa_required_action": null,
      "cisa_exploit_added": null,
      "source_identifier": "security@android.com",
      "affected_products": [
        {
          "id": 2,
          "name": "Android",
          "vendor": "Google",
          "get_absolute_url": "/vuln/product/2/googleandroid/",
          "vendor_part": "o"
        }
      ]
    },
    {
      "id": "CVE-2018-9353",
      "title": "CVE-2018-9353 - In ihevcd_parse_slice_data of ihevcd_parse_slice.c",
      "description": "In ihevcd_parse_slice_data of ihevcd_parse_slice.c there is a possible heap buffer out of bound read due to a missing bounds check. This could lead to remote denial of service with no additional execution privileges needed. User interaction is needed for exploitation.",
      "published": "2024-11-27T23:15:04.560000Z",
      "last_modified": "2024-11-29T22:15:06.027000Z",
      "cvss_score": 6.5,
      "severity": "MEDIUM",
      "cvss_version": "cvss31",
      "epss_scores": [],
      "url": "https://cvefeed.io/vuln/detail/CVE-2018-9353",
      "cisa_required_action": null,
      "cisa_exploit_added": null,
      "source_identifier": "security@android.com",
      "affected_products": [
        {
          "id": 2,
          "name": "Android",
          "vendor": "Google",
          "get_absolute_url": "/vuln/product/2/googleandroid/",
          "vendor_part": "o"
        }
      ]
    },
    {
      "id": "CVE-2018-9352",
      "title": "CVE-2018-9352 - In ihevcd_allocate_dynamic_bufs of ihevcd_api.c th",
      "description": "In ihevcd_allocate_dynamic_bufs of ihevcd_api.c there is a possible resource exhaustion due to integer overflow. This could lead to remote denial of service with no additional execution privileges needed. User interaction is needed for exploitation.",
      "published": "2024-11-27T23:15:04.467000Z",
      "last_modified": "2024-11-29T22:15:05.890000Z",
      "cvss_score": 6.5,
      "severity": "MEDIUM",
      "cvss_version": "cvss31",
      "epss_scores": [],
      "url": "https://cvefeed.io/vuln/detail/CVE-2018-9352",
      "cisa_required_action": null,
      "cisa_exploit_added": null,
      "source_identifier": "security@android.com",
      "affected_products": [
        {
          "id": 2,
          "name": "Android",
          "vendor": "Google",
          "get_absolute_url": "/vuln/product/2/googleandroid/",
          "vendor_part": "o"
        }
      ]
    },
    {
      "id": "CVE-2018-9351",
      "title": "CVE-2018-9351 -  Huawei HEVC/EVC Denial of Service Out-of-Bounds Read",
      "description": "In ih264e_fmt_conv_420p_to_420sp of ih264e_fmt_conv.c there is a possible out of bound read due to missing bounds check. This could lead to remote denial of service with no additional execution privileges needed. User interaction is needed for exploitation.",
      "published": "2024-11-27T23:15:04.343000Z",
      "last_modified": "2024-11-29T22:15:05.750000Z",
      "cvss_score": 6.5,
      "severity": "MEDIUM",
      "cvss_version": "cvss31",
      "epss_scores": [],
      "url": "https://cvefeed.io/vuln/detail/CVE-2018-9351",
      "cisa_required_action": null,
      "cisa_exploit_added": null,
      "source_identifier": "security@android.com",
      "affected_products": [
        {
          "id": 2,
          "name": "Android",
          "vendor": "Google",
          "get_absolute_url": "/vuln/product/2/googleandroid/",
          "vendor_part": "o"
        }
      ]
    },
    {
      "id": "CVE-2018-9350",
      "title": "CVE-2018-9350 - In ih264d_assign_pic_num of ih264d_utils.c there i",
      "description": "In ih264d_assign_pic_num of ih264d_utils.c there is a possible out of bound read due to missing bounds check. This could lead to a denial of service with no additional execution privileges needed. User interaction is needed for exploitation.",
      "published": "2024-11-27T22:15:05.197000Z",
      "last_modified": "2024-11-29T22:15:05.603000Z",
      "cvss_score": 6.5,
      "severity": "MEDIUM",
      "cvss_version": "cvss31",
      "epss_scores": [],
      "url": "https://cvefeed.io/vuln/detail/CVE-2018-9350",
      "cisa_required_action": null,
      "cisa_exploit_added": null,
      "source_identifier": "security@android.com",
      "affected_products": [
        {
          "id": 2,
          "name": "Android",
          "vendor": "Google",
          "get_absolute_url": "/vuln/product/2/googleandroid/",
          "vendor_part": "o"
        }
      ]
    },
    {
      "id": "CVE-2018-9349",
      "title": "CVE-2018-9349 - \"Movian mcomp out-of-bounds read vulnerability\"",
      "description": "In mv_err_cost of mcomp.c there is a possible out of bounds read due to missing bounds check. This could lead to denial of service with no additional execution privileges needed. User interaction is needed for exploitation.",
      "published": "2024-11-27T22:15:05.090000Z",
      "last_modified": "2024-11-29T22:15:05.460000Z",
      "cvss_score": 6.5,
      "severity": "MEDIUM",
      "cvss_version": "cvss31",
      "epss_scores": [],
      "url": "https://cvefeed.io/vuln/detail/CVE-2018-9349",
      "cisa_required_action": null,
      "cisa_exploit_added": null,
      "source_identifier": "security@android.com",
      "affected_products": [
        {
          "id": 2,
          "name": "Android",
          "vendor": "Google",
          "get_absolute_url": "/vuln/product/2/googleandroid/",
          "vendor_part": "o"
        }
      ]
    },
    {
      "id": "CVE-2024-9369",
      "title": "CVE-2024-9369 - Google Chrome Renderer Out-of-Bounds Write",
      "description": "Insufficient data validation in Mojo in Google Chrome prior to 129.0.6668.89 allowed a remote attacker who had compromised the renderer process to perform an out of bounds memory write via a crafted HTML page. (Chromium security severity: High)",
      "published": "2024-11-27T18:15:18.560000Z",
      "last_modified": "2024-11-27T19:15:33.883000Z",
      "cvss_score": 5.5,
      "severity": "MEDIUM",
      "cvss_version": "cvss31",
      "epss_scores": [],
      "url": "https://cvefeed.io/vuln/detail/CVE-2024-9369",
      "cisa_required_action": null,
      "cisa_exploit_added": null,
      "source_identifier": "chrome-cve-admin@google.com",
      "affected_products": [
        {
          "id": 573,
          "name": "Chrome",
          "vendor": "Google",
          "get_absolute_url": "/vuln/product/573/googlechrome/",
          "vendor_part": "a"
        },
        {
          "id": 8224,
          "name": "Edge chromium",
          "vendor": "Microsoft",
          "get_absolute_url": "/vuln/product/8224/microsoftedge_chromium/",
          "vendor_part": "a"
        }
      ]
    }
  ]
}
```
